servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/HostHeaderHttpRequesterFilter.java

```diff
@@ -35,25 +36,39 @@
 import static io.servicetalk.http.api.HttpHeaderNames.HOST;
 import static io.servicetalk.http.api.HttpProtocolVersion.HTTP_1_0;
 import static io.servicetalk.utils.internal.NetworkUtils.isValidIpV6Address;
+import static java.util.Objects.requireNonNull;
 
 /**
  * A filter which will set a {@link HttpHeaderNames#HOST} header with the fallback value if the header is not already
  * present in {@link HttpRequestMetaData}.
  */
 public final class HostHeaderHttpRequesterFilter implements StreamingHttpClientFilterFactory,
                                                             StreamingHttpConnectionFilterFactory {
+
+    // TODO: docs, and this feels hacky.
+    public static final ContextMap.Key<CharSequence> AUTHORITY_KEY =
+            ContextMap.Key.newKey("RequestTarget", CharSequence.class);
```

Comment:
Alternatively, we could make the class extendable with the `setHostHeader` method overridable and make a private extension in the http-netty project. That doesn't change the hack but at least it doesn't leak it.

Reply:
Ideally we could have simply consumed the host and port from the `HttpRequestMetaData` but it turns out that is overwritten in the `DefaultMultiAddressUrlHttpClientBuilder`.